cmd/argocd-server/commands/root.go

```diff
@@ -98,6 +100,7 @@ func NewCommand() *cobra.Command {
 	command.Flags().BoolVar(&insecure, "insecure", false, "Run server without TLS")
 	command.Flags().StringVar(&staticAssetsDir, "staticassets", "", "Static assets directory path")
 	command.Flags().StringVar(&baseHRef, "basehref", "/", "Value for base href in index.html. Used if Argo CD is running behind reverse proxy under subpath different from /")
+	command.Flags().StringVar(&rootPath, "rootpath", "", "Used if Argo CD is running behind reverse proxy under subpath different from /")
```

Comment:
Argo CD CLI has an option to talk to the server using grpc web protocol (e.g. `argocd app list --grpc-web`). 

The grpc web requests should use root path, so we need to add CLI flag to specify path. Can you please add `--grpc-web-root-path` flag. If flag is set then CLI should send grpc web requests with the specified path. Request URL is constructed here:


https://github.com/argoproj/argo-cd/blob/4bd81730ef77301e50ab19568fe7419160bbb9c9/pkg/apiclient/grpcproxy.go#L58


For convenience we should assume that user wants to use grpc web if either `--grpc-web` or `--grpc-web-root-path` is set